partner_internal_code/models/res_partner.py

```diff
@@ -14,7 +14,7 @@ class Partner(models.Model):
         copy=False,
     )
 
-    @api.model
+    @api.model_create_multi
     def create(self, vals):
         if not vals.get('internal_code', False):
```

Comment:
va a seguir generando error. porque vals es un array de diccionarios con los datos a crear
Creo que el código tendría que ser mas parecido a esto: 

```
@api.model_create_multi
    def create(self, vals_list):
        for vals in vals_list:
          if not vals.get('internal_code'):
              vals['internal_code'] = self.env['ir.sequence'].next_by_code( 'partner.internal.code')
        return super().create(vals_list)
             
```
Ademas un par de cosas:
- Por convención  el argumento que usamos en model_create_multi es **vals_list** no **vals**. Si bien funcionalmente no cambia nada, cuando alguien lee el código, sabe que se refiere a un array de diccionarios y no a un diccionario.
- En el contexto de un *if not* no es necesario definir un valor por defecto de la función *get()*. Porque si el valor no existe  es **None** y en este contexto se  comporta igual que un False
-  En pos de la legibilidad de la función por ahí es mejor hacerla así (creo que get no va a ejecutar next_by_code si existe internal_code pero probalo )

```
@api.model_create_multi
    def create(self, vals_list):
        for vals in vals_list:
              vals['internal_code'] =  vals.get('internal_code', self.env['ir.sequence'].next_by_code( 'partner.internal.code'))
        return super().create(vals_list)
```

Reply:
@maq-adhoc Filo, estuve probando lo que me dijiste. Buenisima la explicación de arriba! Lamentablemente get no ejecuta next_by_code por lo que lo deje con el if not. Cualquier cosa avisame pero ya estaría este método corregido y funcionando